Uploaded lesson 4 homework

ErnestMihasenko opened the pull request:
Я наверное не совсем понял самое начало домашнего задания: "Создать массив из кортежей (Int, String)"
Массив же может содержать данные только одного типа. Если что-то неверно, поясни, пожалуйста, буду дорабатывать :)

ErnestMihasenko committed:
Uploaded lesson 4 homework

Я наверное не совсем понял самое начало домашнего задания: "Создать массив из кортежей (Int, String)"
Массив же может содержать данные только одного типа. Если что-то неверно, поясни, пожалуйста, буду дорабатывать :)

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+import UIKit
+
+var arrayNumbers: [Int] = [-1, 2, 6, 12, 0, 5, 7, 8]
```
We should create an array of tuples `(Int, String)`. Yes, an array can only contain elements of one type but `(Int, String)` is a type so you can create an array for it

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+
+var arrayNumbers: [Int] = [-1, 2, 6, 12, 0, 5, 7, 8]
+
+for elements in 0...arrayNumbers.count - 1 {
```
There's nothing wrong with closed range like this (0... arrayNumbers.count - 1) but you can optimise this line using simple range 0..< arrayNumbers.count

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+var arrayNumbers: [Int] = [-1, 2, 6, 12, 0, 5, 7, 8]
+
+for elements in 0...arrayNumbers.count - 1 {
+    let powNumbers = Int(pow(Double(arrayNumbers[elements]), Double(2)))
```
There's no need to cast `2` to `Double` explicitly, you can just leave `2` here and the compilator will treat it as `Double` type

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+
+for elements in 0...arrayNumbers.count - 1 {
+    let powNumbers = Int(pow(Double(arrayNumbers[elements]), Double(2)))
+    print(powNumbers)
```
You need to put `powNumber` in a new array

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+    let powNumbers = Int(pow(Double(arrayNumbers[elements]), Double(2)))
+    print(powNumbers)
+}
+// Вопрос по этому пункту. Как мы можем использовать данные массива powNumbers, если, например, потребуется вызвать их в другой функции?
```
You can just pass this array to this function

darya-gurinovich commented on Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,21 @@
+
+for evenNumbers in arrayNumbers {
+    if evenNumbers % 2 == 0 {
+        print(evenNumbers)
```
You also need to put filtered elements in a new array

darya-gurinovich reviewed (changes requested):
Good effort. Please, create the array of tuples and store your cycles results in new arrays

ErnestMihasenko committed:
Updated homework 4 

I deleted previous files, and uploaded project again. Is it correct, or should I better leave old files?

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+import UIKit
+
+let firstTupple: (Int, String) = (0, "zero")
+let secondTupple = (2, "two")
+let thirdTupple = (3, "three")
+let fourthTupple = (4, "four")
+let fifthTupple = (5, "five")
+let sixthTupple = (6, "six")
+let seventhTupple = (7, "seven")
+let eigthsTupple = (8, "eigth")
```
You have a misspelling. Tuple should be written with one p

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+let seventhTupple = (7, "seven")
+let eigthsTupple = (8, "eigth")
+
+var arrayTupples = [firstTupple, secondTupple, thirdTupple, fourthTupple, fifthTupple, sixthTupple, seventhTupple, eigthsTupple]
```
Actually, you can create all tuples here without creating separate variables. It's not wrong to do it like you did, it's just a possible improvement

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+let eigthsTupple = (8, "eigth")
+
+var arrayTupples = [firstTupple, secondTupple, thirdTupple, fourthTupple, fifthTupple, sixthTupple, seventhTupple, eigthsTupple]
+var powArrayTupples = arrayTupples
```
What's the point to create this variable? You can just use `arrayTupples` inside the cycle

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+var arrayTupples = [firstTupple, secondTupple, thirdTupple, fourthTupple, fifthTupple, sixthTupple, seventhTupple, eigthsTupple]
+var powArrayTupples = arrayTupples
+
+for elements in 0..<powArrayTupples.count {
```
It's better to rename `elements` to `index` because you're looping through elements indices here

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+var powArrayTupples = arrayTupples
+
+for elements in 0..<powArrayTupples.count {
+    let powNumbersArray = [(Int(pow(Double(powArrayTupples[elements].0), 2)), powArrayTupples[elements].1)] as [Any]
+    print(powNumbersArray)
```
If you want just to print tuple here, there's no need to create an array here. Also there's no need to cast it to `[Any]`. If you want to store this new tuple in an array, you need to create a variable before and append a new value here

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+    print(powNumbersArray)
+}
+
+var evenArrayTupples = arrayTupples
```
The same about necessity of this variable creation. You can just use `arrayTupples` inside the cycle as well

darya-gurinovich commented on Lesson4HomeWork.playground/Lesson4HomeWork.playground/Contents.swift:
```diff
@@ -0,0 +1,31 @@
+
+var evenArrayTupples = arrayTupples
+var evenNumberArray = [Int]()
+for evenNumbers in 0..<evenArrayTupples.count {
```
We need to go through elements not through elements indices in this task

darya-gurinovich reviewed (approved):
Good work! Just a couple of small improvements here

darya-gurinovich commented:
I've seen that you've left a comment that you've deleted old solution files. But I can still see them here. Please, check it
